contracts/rewards/src/contract/execute.rs

```diff
@@ -64,10 +65,30 @@ where
 
     pub fn update_params(
         &mut self,
-        _new_params: RewardsParams,
-        _block_height: u64,
+        new_params: RewardsParams,
+        block_height: u64,
     ) -> Result<(), ContractError> {
-        todo!()
+        let mut cur_epoch = self.get_current_epoch(block_height)?;
+        // If the param update reduces the epoch duration such that the current epoch immediately ends,
+        // start a new epoch at this block, incrementing the current epoch number by 1.
+        // This prevents us from jumping forward an arbitrary number of epochs, and maintains consistency for past events.
+        // (i.e. we are in epoch 0, which started at block 0 and epoch duration is 1000. At epoch 500, the params
+        // are updated to shorten the epoch duration to 100 blocks. We set the epoch number to 1, to prevent skipping
+        // epochs 1-4, and so all events prior to the start of epoch 1 have an epoch number of 0)
+        if cur_epoch.block_height_started + u64::from(new_params.epoch_duration) < block_height {
+            cur_epoch = Epoch {
+                block_height_started: block_height,
+                epoch_num: cur_epoch.epoch_num + 1,
+                rewards: new_params.rewards_per_epoch,
+            };
+        } else {
+            cur_epoch.rewards = new_params.rewards_per_epoch;
+        }
```

Comment:
```suggestion
        let should_end = cur_epoch.block_height_started + u64::from(new_params.epoch_duration) < block_height;
        let cur_epoch =
            if  should_end{
                Epoch {
                    block_height_started: block_height,
                    epoch_num: cur_epoch.epoch_num + 1,
                    rewards: new_params.rewards_per_epoch,
                }
            } else {
                Epoch {                    
                    rewards: new_params.rewards_per_epoch,
                    ..cur_epoch
                }
            };
```
this way both branches are on equal footing, the whole thing is an expression and you don't need to mutate state